### PR TITLE
Make log test more resilient

### PR DESCRIPTION
**Pull request by sebsoto**
Adds a retry for node log retrieval, and logs the command output upon
failure.

### CODE REVIEW

File: test/e2e/logs_test.go [hunk 1]
```diff
@@ -35,12 +39,27 @@ func testNodeLogs(t *testing.T) {
...
+					}
+					return true, nil
+				})
+				assert.NoError(t, err)
 			})
 		}
 		// Grab the optional logs for debugging purposes
 		for _, file := range optionalLogs {
-			_ = retrieveLog(node.Name, file, nodeDir)
+			_ = wait.PollImmediate(retry.Interval, retry.ResourceChangeTimeout, func() (bool, error) {
```
Comment (aravindhp) [inline test/e2e/logs_test.go:55]: Why are we ignoring the error here?

Comment (sebsoto) [reply test/e2e/logs_test.go:55]: These logs are not guaranteed to be present. We still want to collect them if they exist

**Review by aravindhp — APPROVED** [review]
Thanks for being proactive and fixing this issue, @sebsoto

File: hack/run-ci-e2e-test.sh [hunk 2]
```diff
@@ -14,7 +14,13 @@ WMCO_PATH_OPTION=""
 export CGO_ENABLED=0
 
 get_WMCO_logs() {
-  oc logs -l name=windows-machine-config-operator -n $WMCO_DEPLOY_NAMESPACE --tail=-1 >> "$ARTIFACT_DIR"/wmco.log
+  retries=0
+  until [ $retries -gt 4 ] || oc logs -l name=windows-machine-config-operator -n $WMCO_DEPLOY_NAMESPACE --tail=-1 >> "$ARTIFACT_DIR"/wmco.log
+  do
+	let retries+=1
+	echo "failed to get WMCO logs"
+	sleep 5
```
Comment (aravindhp) [inline hack/run-ci-e2e-test.sh:22]: Is it possible to indent?

**Comment by sebsoto** [comment]
retest

File: test/e2e/logs_test.go (hunk 1 above)
Comment (aravindhp) [reply test/e2e/logs_test.go:55]: Please add a comment stating this. `Grab the optional logs for debugging purposes` does not explain why we are not looking at the error.

**Review by aravindhp — APPROVED** [review]

File: hack/run-ci-e2e-test.sh (hunk 2 above)
Comment (sebsoto) [reply hack/run-ci-e2e-test.sh]: I just replaced the tabs with spaces, didnt realize they were there.
 This file is split between 4 and 2 spaces, but the majority of it is 2 space indenting, so chose 2 for this function.

**Pull request merged**